models/model.js

```diff
@@ -1,22 +1,29 @@
-const db = require('../db/connection.js');
+const db = require("../db/connection.js");
 const fs = require("fs/promises");
 
-function selectAllTopics (){
-
-return db.query(`SELECT * FROM topics`).then((topics) => {
-    if (topics.rows.length === 0){
-        return Promise.reject({status: 404, msg: 'Not Found'})
+function selectAllTopics() {
+  return db.query(`SELECT * FROM topics`).then((topics) => {
+    if (topics.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not Found" });
     }
-   
-return topics.rows
-})
 
+    return topics.rows;
+  });
+}
+
+function selectAllEndpoints() {
+  return fs.readFile(`endpoints.json`, "utf8");
 }
 
-function selectAllEndpoints(){
- return fs.readFile(`endpoints.json`, "utf8")
- }
- 
- 
-module.exports = {selectAllTopics, selectAllEndpoints}
+function selectArticle(id) {
+  return db.query(`SELECT articles.author, articles.title, articles.article_id, articles.body, articles.topic, articles.created_at,    articles.votes, articles.article_img_url
+  FROM articles 
+  WHERE articles.article_id=$1`, [id]).then((response) => {
+    if (response.rows.length ===0){
+        return Promise.reject({status: 404, msg: 'Not Found'})
+    }
+  return response.rows
```

Comment:
`rows` is an array. We only want the very first (and only) item in our array so we can send back the article object, not an array